routes/articles.js

```diff
@@ -0,0 +1,129 @@
+const express = require('express')
+let router = express.Router()
+const Article = require('../models/article')
+const {isLoggedIn, checkArticleOwnership} = require('../middleware')
+
+function escapeRegex(text) {
+    return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+}
+
+//INDEX ROUTE -- Show all articles
+router.get('/', (req, res) => {
+    if(req.query.search && req.xhr) {
+        const regex = new RegExp(escapeRegex(req.query.search), 'gi')
+        Article.find({name: regex}, (err, articles) => {
+            if(err) {
+                req.flash('Oops! Something went wrong!')
+                console.log('bad params, Article - INDEX ROUTE')
+                return res.redirect('/')
+            }
+            return res.status(200).json(articles)
+        })
+    }
+
+    Article.find({}, (err, articles) => {
+        res.render('articles/index', {articles, currentUser: req.user, page: 'articles'})
+    })
+})
+
+//CREATE ROUTE
+router.post('/', isLoggedIn, (req, res) => {
+    if(!__verifyParams(req.body)) {
+        req.flash('Oops! Something went wrong!')
+        console.log('bad params, Article - CREATE ROUTE')
+        return res.redirect('/articles')
+    }
+
+    const {title, description, body} = req.body
+    const author = {id: req.user._id, username: req.user.username}
+
+    Article.create({author, title, description, body}, err => {
+        if(err) throw err
+
+        req.flash('success', 'Article created!')
+        res.redirect('/articles')
+    })
+})
+
+//NEW - Show form to create new article
+router.get('/new', isLoggedIn, (req, res) => {
+    res.render('articles/new.ejs')
+})
+
+//SHOW - Show more info about one article
+router.get('/:id', (req, res) => {
+    if(req.params.id === undefined) {
+        req.flash('error', 'Oops! Something went wrong!')
+        return res.redirect('/articles')
+    }
+
+    Article.findById(req.params.id).populate('comments').exec((err, article) => {
```

Comment:
What are we doing here? `populate` seems like dev code

Reply:
https://mongoosejs.com/docs/populate.html